### ISSUE TITLE
Need Help plz

### ISSUE THREAD

**Issue by JesterDoobiesmoker** [greenlover1991/pebble-remote-for-automate#3]
I came here from the Pebble/Rebble Appstore, I don't have an email app on my phone so can't contact you thru their Store. My issue is that I cannot figure out how to use your app, it shows many Steves  on my watch and abcd options but nothing works. Appstore description mentions needed flows or files of some sort but I can't find them anywhere. My phone is a Galaxy S4 running an AOSP-based version of Android 5.0.2, rooted.Thanks for your time

**Comment by greenlover1991** [comment]
Woops. I forgot to add instructions in the repo's readme. 

Steps:
1. Download Llamalab's Automate

**Comment by greenlover1991** [comment]
Woops. I forgot to add instructions in the repo's readme. Weird, I remember composing a long instruction manual somewhere...

----



Basically, you need the Automate app as the communication bridge to your Pebble, and to setup triggers/actions. The parameters of the trigger is in the Automate flow description, which is sent by the Listing flow. The triggering flow listens for the flow tapped by the user from the Pebble app. 

**Steps:**
1. Download Llamalab's Automate at https://play.google.com/store/apps/details?id=com.llamalab.automate. It's similar to Tasker. 
2. Download the ff. flows:
 * Listing flow = https://llamalab.com/automate/community/flows/21539
 * Triggering flow = https://llamalab.com/automate/community/flows/21540
 * Sample triggering flow - Turn onFlashlight = https://llamalab.com/automate/community/flows/35568
 * (Optional) Sample triggering flow with Pebble 2 HR voice input as payload to the flow (speak "Boy" to your watch, and the phone will speak back) = https://llamalab.com/automate/community/flows/35569
3. After downloading the app and the flows, start the listing flow and the triggering flow. The listing flow will report to the pebble app the list of available flows that can be triggered. The triggering flow waits for button clicks from the pebble app.  
4. In the pebble app, on tap of the options, the triggering flow will reroute to the actual flow to be executed. 

**Notes/disclaimer:**
1. You may have to edit the listing flow to point to the correct flow ID of the downloaded triggering and sample flows.
2. The free Automate app has a limit of 30 running blocks. So you may have to adjust the flow as well if the setup doesn't work. I'm currently a premium Automate user which removes the limit. 
3. I haven't used my Pebble in 2 years, so I can't say if this app still works :)

Lemme know if that helps. Thanks

**Comment by Swaddy33** [comment]
>     3. I haven't used my Pebble in 2 years, so I can't say if this app still works :)
> Lemme know if that helps. Thanks

Well this helps a lot, but I still couldn't get it running.
I tried to troubleshoot a bit and the only problem I see is the received "msg_data" which contains a wrong flow-id (example: 3355697 instead of 143). I know that the data array in the list generator is fine, so the numberchange might happen when the pebble app receives the data...

It would be awesome for the pebble Community to have this app working. Please consider fixing it (I would even spend you a drink :-) )

**Comment by greenlover1991** [comment]
> I know that the data array in the list generator is fine

Just to confirm, you were able to list down all the Automate Flows in your Pebble watch?

> the only problem I see is the received "msg_data" which contains a wrong flow-id (example: 3355697 instead of 143)

Hmmm so your main issue is the flow you selected in your watch is not triggered in the Automate app?

Have you tried hardcoding the flow ID first? The flow ID corresponds the Flow URI in your Automate app, *not* the community URL. So if it's a freshly installed app, most likely the first flow would have the ID of 1. 

Can you confirm if the payload from the broadcast received in the "Pebble: Trigger flows" with the action `com.getpebble.action.app.RECEIVE` is what you expected? That is, use a `Log append` block to print the `json` variable, and reply here the output. 

Did you modify the source code or the flow blocks?

----

> It would be awesome for the pebble Community to have this app working. Please consider fixing it (I would even spend you a drink :-) )

@Swaddy33 haha. thanks for the offer. I'd gladly accept a new Pebble instead :D kidding aside, I need to locate my P2HR first.

**Comment by Swaddy33** [comment]
> Just to confirm, you were able to list down all the Automate Flows in your Pebble watch?

Yes, it lists the ones with pbv=1 like it should. Sometimes I need to open the app twice because it closes immediately but that's not my main concern yet ;-)

> Hmmm so your main issue is the flow you selected in your watch is not triggered in the Automate app?

Exactly because the flowid is wrong.
 
> Have you tried hardcoding the flow ID first? The flow ID corresponds the Flow URI in your Automate app, _not_ the community URL. So if it's a freshly installed app, most likely the first flow would have the ID of 1.

I hardcoded my flashlight-id 143 into the trigger flow and it executed as it should!
 
> Can you confirm if the payload from the broadcast received in the "Pebble: Trigger flows" with the action `com.getpebble.action.app.RECEIVE` is what you expected? That is, use a `Log append` block to print the `json` variable, and reply here the output.

This is what I receive from the broadcast:
_10-11 12:47:25.047 U 6534@61: msg_data as String: [{"key":10001,"type":"string","length":0,"value":"REQ_TRIGGER_FLOW"},{"key":10002,"type":"string","length":0,"value":"3355697"}], transaction_id as Int: 2_
 
> Did you modify the source code or the flow blocks?

Modifying the source code is way over my head and I only added "log append" to the flow to find the problem. And like you asked (and mentioned before) I tried hardcoding the flow-id in a copy of the triggerflow.
 
> @Swaddy33 haha. thanks for the offer. I'd gladly accept a new Pebble instead :D kidding aside, I need to locate my P2HR first.

Well I hope you'll find it :-D
Actually some are selling their pebble around 30$ online...

**Comment by greenlover1991** [comment]
> This is what I receive from the broadcast:
10-11 12:47:25.047 U 6534@61: msg_data as String: [{"key":10001,"type":"string","length":0,"value":"REQ_TRIGGER_FLOW"},{"key":10002,"type":"string","length":0,"value":"3355697"}], transaction_id as Int: 2

@Swaddy33 Thanks. I found my P2HR (surprisingly it can still hold a charge) and am able to replicate the issue.

**Cause:** 
* In `Pebble: Flow list generate` flow, the flow ID is correctly sent to the watch app, but with incorrect type `string` instead of `int`, which caused the watch app's C code to treat it as overflow/garbage value(?) and changed it to `3355697`.

**Fix:**
1. In `Pebble: Flow list generate` flow, find the `Variable set` block with variable `data` (block #46), and modify the `type` of `10003` from `string` to `int` , and modify the `length` from `0` to `4`. So it should look like:
```javascript
// old:
[{"key": 10003, "type": "string", "length": 0, "value": flow["_id"]}, ...]
```

```javascript
// new:
[{"key": 10003, "type": "int", "length": 4, "value": flow["_id"]}, ...]
```
2. restart the flow
3. restart the pebble watch app
4. Tap the selected flow in the watch app, and the flow should be triggered properly.

I uploaded the fix in the community flows in https://llamalab.com/automate/community/flows/21539 if you don't want to manually update the flow. 

I tested both Start and Stop buttons, and they're now working.
Let me know if it's fixed on your side. Thanks.

**Comment by Swaddy33** [comment]
Oh nooo, it was just the wrong data-type?!
I can confirm the flows work like a charm!

But now there is a problem with the app.
- When I open it, the trigger flow sends the list to the watch and the app displays the list
- I go back and open the app again. The screen stays mostly blank, only the time is displayed on the top
- I need to force close the app and wait to try again

First I thought it might be because of a very long description in a flow, so I removed all flows bar the demo onces but the problem remains :-/
Here's a little clip how it looks like:
[remote-open.zip](https://github.com/greenlover1991/pebble-remote-for-automate/files/5373348/remote-open.zip)
